tidy3d/components/data/monitor_data.py

```diff
@@ -1126,6 +1127,18 @@ def _isel(self, **isel_kwargs):
         update_dict = {key: field.isel(**isel_kwargs) for key, field in update_dict.items()}
         return self._updated(update=update_dict)
 
+    def _assign_coords(self, **isel_kwargs):
+        """Wraps ``xarray.DataArray.assign_coords`` for all data fields that are defined over frequency and
+        mode index. Used in ``overlap_sort`` but not officially supported since for example
+        ``self.monitor.mode_spec`` and ``self.monitor.freqs`` will no longer be matching the
+        newly created data."""
+
+        update_dict = dict(self._grid_correction_dict, **self.field_components)
```

Comment:
_grid_correction_dict is a property so it shouldn't be added here at all?

Reply:
I guess this is adding `grid_primal_correction` and `grid_dual_correction` to the update list. I copied this from `_isel`, but it is probably not needed as these fields keep their old values...

Reply:
oh actually, it does do something -- it updates the coordinates on those two fields as well as the field_components. This may actually be needed so best to leave it.

Reply:
Ah, got it.